test/controllers/income/estimatedPay/update/IncomeUpdateCalculatorControllerSpec.scala

```diff
@@ -406,26 +407,24 @@ class IncomeUpdateCalculatorControllerSpec
   }
 
   "bonusPaymentsPage" must {
-    "display bonusPayments" when {
-      "journey cache returns employment name and id" in {
-        val testController = createTestIncomeUpdateCalculatorController
-        when(testController.journeyCacheService.currentValue(Matchers.eq(UpdateIncome_PayslipDeductionsKey))(any())).thenReturn(Future.successful(None))
-        val result = testController.bonusPaymentsPage()(RequestBuilder.buildFakeRequestWithAuth("GET"))
-        status(result) mustBe OK
+    "display bonusPayments" in {
+      val testController = createTestIncomeUpdateCalculatorController
+      when(testController.journeyCacheService.mandatoryValues(any())(any())).thenReturn(Future.successful(Seq("1", "employer1")))
+      val result = testController.bonusPaymentsPage()(RequestBuilder.buildFakeRequestWithAuth("GET"))
+      status(result) mustBe OK
 
-        val doc = Jsoup.parse(contentAsString(result))
-        doc.title() must include(messages("tai.bonusPayments.title"))
-      }
+      val doc = Jsoup.parse(contentAsString(result))
+      doc.title() must include(messages("tai.bonusPayments.title", TaxYearRangeUtil.currentTaxYearRangeBetweenDelimited))
```

Comment:
'Nice to have' Could replace the checking of the heading to the full rendering of the view